### PR TITLE
Fix #61

### LINKED ISSUE #59: Update docker-compose to no rely on manual env vars export
Yet the command should be kept in the README as it is useful for running the development server.

- Update docker-compose.yml
- Update the README

### LINKED ISSUE #60: Fix env vars export in CI/CD
Apply fix implemented in https://github.com/nlpsandbox/data-node

### LINKED ISSUE #61: Update npm dependencies
A new major version of `@openapitools/openapi-generator-cli` is available

### PR DESCRIPTION
**Pull request by tschaffter**
- Fixes #59 
- Fixes #60 
- Fixes #61

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,42 +16,38 @@ API service and web client of the Registry of Open Community Challenges
 
 1. Create the file that contains the future environment variables
 
-       cp .env.sample .env
+       cp .env.example .env
```
Comment (vpchung) [inline README.md:19]: Would it be helpful to explicitly mention that the file will work as-is after copying?  If they _wanted_ to, they can edit the environment variables, but it is not necessary.

File: README.md [hunk 2]
```diff
@@ -16,42 +16,38 @@ API service and web client of the Registry of Open Community Challenges
...
 
-2. Export the variables defined in *.env* to environment variables
-
-       export $(grep -v '^#' .env | xargs -d '\n')
-
-3. Start the Data Node API service
+2. Start the Data Node API service
```
Comment (vpchung) [inline README.md:21]:
```suggestion
2. Start the ROCC API service
```

File: README.md [hunk 3]
```diff
@@ -16,42 +16,38 @@ API service and web client of the Registry of Open Community Challenges
...
 
-       docker-compose up
+       docker-compose up --build
```
Comment (vpchung) [inline README.md:23]: We could add a note that if they want the service to run in the background, they could add the `-d` or `--detach` flag to the command.

File: README.md [hunk 4]
```diff
@@ -16,42 +16,38 @@ API service and web client of the Registry of Open Community Challenges
...
 
-4. Go to the API service documentation page <http://localhost:8080/api/v1/ui/>
+3. Go to the API service documentation page <http://localhost:8080/api/v1/ui/>
```
Comment (vpchung) [inline README.md:25]: I think we should also add a step after this on how to stop the service:

"To stop the API service, enter `Ctrl+C` followed by `docker-compose down`.  If running in detached mode, you will only need to enter `docker-compose down`."

File: docker-compose.yml [hunk 5]
```diff
@@ -3,39 +3,44 @@ version: "3.8"
 services:
   db:
     image: sagebionetworks/mongo:4.4.3
-    container_name: rocc-db
+    container_name: data-node-db
```
Comment (vpchung) [inline docker-compose.yml:6]:
```suggestion
    container_name: rocc-db
```

File: docker-compose.yml [hunk 6]
```diff
@@ -3,39 +3,44 @@ version: "3.8"
...
+      - MONGO_INITDB_DATABASE=${MONGO_INITDB_DATABASE}
+      - MONGO_USERNAME=${MONGO_USERNAME}
+      - MONGO_PASSWORD=${MONGO_PASSWORD}
     volumes:
       - database-data:/data/db
     ports:
       - "${MONGO_PORT}:27017"
 
-  api:
+  data-node:
```
Comment (vpchung) [inline docker-compose.yml:19]:
```suggestion
  api:
```

**Review by vpchung — CHANGES_REQUESTED** [review]
Not sure if you needed the review, but I did one! 

Mostly just had a couple suggestions for the README; otherwise, looks good!

File: README.md (hunk 3 above)
Comment (tschaffter) [reply README.md:23]: Done in next commit

File: README.md (hunk 1 above)
Comment (tschaffter) [reply README.md:19]: Done in next commit.

File: README.md (hunk 4 above)
Comment (tschaffter) [reply README.md:25]: Done in next commit.

File: docker-compose.yml (hunk 6 above)
Comment (tschaffter) [reply docker-compose.yml:19]: Renamed to `rocc`.

**Pull request merged**